### PR TITLE
Revise the discussion about how plugins work in the documentation.

### PR DESCRIPTION
**Pull request by bangerth**
While finishing up #1775.

### CODE REVIEW

File: doc/sphinx/user/extending/idea-of-plugins.md [hunk 1]
```diff
@@ -38,30 +54,25 @@ The way this is achieved is through the following two steps:
...
+    If you want to implement a new model for gravity, in its simplest form
+    you just need to write a
     class that derives from this base class and implements the
     `gravity_vector` function. If your model wants to read parameters from the
-    input file, you also need to have functions called `declare_parameters`
-    and `parse_parameters` in your class with the same signatures as the ones
-    above. On the other hand, if the new model does not need any run-time
+    input file, you also need to have functions called `declare_parameters()`
+    and `parse_parameters()` in your class with the same signatures as the ones
+    declared in the `Plugins::Interface` base class.
```
Comment (gassmoeller) [inline doc/sphinx/user/extending/idea-of-plugins.md:71]:
```suggestion
    declared in the `Plugins::InterfaceBase` base class.
```

Comment (gassmoeller) [inline doc/sphinx/user/extending/idea-of-plugins.md:71]:
```suggestion
    declared in the `Plugins::InterfaceBase` base class.
```

**Review by gassmoeller — APPROVED** [review]
Yes, thanks for the update. Feel free to merge yourself after addressing my comment.

**Comment by bangerth** [comment]
Thanks for fixing the typo yourself :-)

**Pull request merged**